File: modules/secure-recorder/ios/RecordingState.swift
import Foundation

/**
 * Immutable recording state (thread-safe)
 */
struct RecordingState {
  let isRecording: Bool
  let sessionId: String?
  let filePath: String?
  
  static let idle = RecordingState(isRecording: false, sessionId: nil, filePath: nil)
}

File: modules/secure-recorder/android/src/main/java/expo/modules/securerecorder/RecordingState.kt
package expo.modules.securerecorder

/**
 * Immutable recording state (thread-safe)
 */
data class RecordingState(
  val isRecording: Boolean,
  val sessionId: String?,
  val filePath: String?
) {
  companion object {
    val IDLE = RecordingState(false, null, null)
  }
}
